be/src/olap/delta_writer.cpp

```diff
@@ -380,6 +382,23 @@ Status DeltaWriter::close_wait(const PSlaveTabletNodes& slave_tablet_nodes,
             _request_slave_tablet_pull_rowset(node_info);
         }
     }
+    // do calc delete bitmap in segments here
+    // generate delete bitmap, build a tmp rowset and load recent segment
+    if (!_tablet->enable_unique_key_merge_on_write()) {
+        return Status::OK();
+    }
+    auto rowset = _rowset_writer->build_tmp();
+    auto beta_rowset = reinterpret_cast<BetaRowset*>(rowset.get());
+    std::vector<segment_v2::SegmentSharedPtr> segments;
+    RETURN_IF_ERROR(beta_rowset->load_segments(&segments));
+    std::shared_lock meta_rlock(_tablet->get_header_lock());
+    // tablet is under alter process. The delete bitmap will be calculated after conversion.
+    if (_tablet->tablet_state() == TABLET_NOTREADY &&
+        SchemaChangeHandler::tablet_in_converting(_tablet->tablet_id())) {
+        return Status::OK();
+    }
+    RETURN_IF_ERROR(_tablet->calc_delete_bitmap(beta_rowset->rowset_id(), segments, nullptr,
```

Comment:
This section of code should be placed before line 364, because `set_txn_related_delete_bitmap`  was called there.

Reply:
fixed